iree/vm/list.c

```diff
@@ -13,19 +13,19 @@
 
 #include "iree/base/tracing.h"
 
-// Size of each iree_vm_value_type_t in bytes.
-static const iree_host_size_t kValueTypeSizes[7] = {
-    0,  // IREE_VM_VALUE_TYPE_NONE
-    1,  // IREE_VM_VALUE_TYPE_I8
-    2,  // IREE_VM_VALUE_TYPE_I16
-    4,  // IREE_VM_VALUE_TYPE_I32
-    8,  // IREE_VM_VALUE_TYPE_I64
-    4,  // IREE_VM_VALUE_TYPE_F32
-    8,  // IREE_VM_VALUE_TYPE_F64
-};
-static_assert(IREE_VM_VALUE_TYPE_COUNT ==
-                  (sizeof(kValueTypeSizes) / sizeof(kValueTypeSizes[0])),
-              "Enum mismatch");
+static uint8_t iree_vm_value_type_size(iree_vm_value_type_t type) {
+  // Size of each iree_vm_value_type_t in bytes. We bitpack these so that we
+  // can do a simple shift and mask to get the size.
+  const uint32_t kValueTypeSizes = (0u << 0) |   // IREE_VM_VALUE_TYPE_NONE
+                                   (1u << 4) |   // IREE_VM_VALUE_TYPE_I8
+                                   (2u << 8) |   // IREE_VM_VALUE_TYPE_I16
+                                   (4u << 12) |  // IREE_VM_VALUE_TYPE_I32
+                                   (8u << 16) |  // IREE_VM_VALUE_TYPE_I64
```

Comment:
looks like something in here is causing a test to fail

https://source.cloud.google.com/results/invocations/1b1df17a-933a-44b2-ac5f-fecf20ea0a59/targets/iree%2Fgcp_ubuntu%2Fbazel%2Flinux%2Fx86-swiftshader%2Fcore%2Fpresubmit/log

```
iree/vm/bytecode_dispatch_test.cc:127: Failure
Failed
Function expected success but failed with error: <vm>:0: FAILED_PRECONDITION; list<i64>.empty.set(41, MAX_INT_PLUS_1).get(41)=MAX_INT_PLUS_1;
[ 0] bytecode list_ops_i64.test_i64:0 iree/vm/test/list_ops_i64.mlir:9:17
[  FAILED  ] VMIRFunctions/VMBytecodeDispatchTest.Check/list_ops_i64_vmfb_test_i64, where GetParam() = list_ops_i64_vmfb_test_i64 (0 ms)
```